x-pack/test/api_integration/apis/ml/system/node_count.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { getCommonRequestHeader } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  async function runRequest(user: USER) {
+    const { body, status } = await supertest
+      .get(`/internal/ml/ml_node_count`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(getCommonRequestHeader('1'));
+    ml.api.assertResponseStatusCode(200, status, body);
+
+    return body;
+  }
+
+  describe('POST ml/ml_node_count', function () {
+    describe('get ml node count', () => {
+      it('should match expected values', async () => {
+        const resp = await runRequest(USER.ML_POWERUSER);
```

Comment:
Is it worth adding a test for a different user?

Reply:
Added in 5f4dff19b22d9cae78d82cb7ad31c77ff14b2043